appveyor.yml

```diff
@@ -1,26 +1,67 @@
 build: false
-shallow_clone: false
-platform: 'x86'
-clone_folder: C:\projects\better-reflection
+clone_depth: 50
+platform:
+  - x64
+environment:
+  matrix:
+    - dependencies: lowest
+      php_version: 7.1
+    - dependencies: highest
+      php_version: 7.1
 
+  project_directory: c:\projects\better-reflection
+  composer_directory: c:\tools\composer
+  composer_executable: c:\tools\composer\composer.phar
+  php_archive_directory: c:\tools\php-archive
+  php_directory: c:\tools\php
+matrix:
+  fast_finish: true
+clone_folder: c:\projects\better-reflection
+cache:
+  - c:\tools\composer
+  - c:\tools\php-archive
+  - '%LOCALAPPDATA%\Composer'
+init:
+  - ps: $Env:PATH = $Env:php_directory + ';' + $Env:composer_directory + ';' + $Env:PATH
+  - ps: $Env:ANSICON = '121x90 (121x90)'
 install:
-  - cinst -y OpenSSL.Light
-  - SET PATH=C:\Program Files\OpenSSL;%PATH%
-  - sc config wuauserv start= auto
-  - net start wuauserv
-  - cinst -y php
-  - cd c:\tools\php71
-  - copy php.ini-production php.ini
-  - echo date.timezone="UTC" >> php.ini
-  - echo extension_dir=ext >> php.ini
-  - echo extension=php_openssl.dll >> php.ini
-  - echo extension=php_mbstring.dll >> php.ini
-  - cd C:\projects\better-reflection
-  - SET PATH=C:\tools\php71;%PATH%
-  - php -r "readfile('http://getcomposer.org/installer');" | php
-  - php composer.phar install --prefer-source --no-interaction
+  # Download requested PHP version
+  - ps: If ((Test-Path $Env:php_archive_directory) -eq $False) { New-Item -Path $Env:php_archive_directory -ItemType 'directory' }
+  - ps: $requested_php_version = (((choco search php --exact --all-versions -r | Select-String -pattern $Env:php_version) -replace '[php|]', '') | %{ New-Object System.Version $_ } | Sort-Object | Select-Object -Last 1).ToString()
+  - ps: $php_version_url = 'http://windows.php.net/downloads/releases/php-' + $requested_php_version + '-nts-Win32-VC14-' + $Env:platform + '.zip'
+  - ps: $php_version_file = $Env:php_archive_directory + '\php-' + $requested_php_version + '.zip'
```

Comment:
@kukulich well aware of that, but it's still less work to do. 

I'm perfectly ok with having a slower CI and less setup steps. 4m is nothing compared to many other projects I work on.